Simplenote/NSString+Simplenote.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+
+// MARK: - Simplenote API
+//
+extension NSString {
+
+    /// Returns the line (range, string) at the specified location
+    ///
+    func line(at location: Int) -> (NSRange, String) {
+        let queryRange = NSRange(location: location, length: .zero)
+        let resultRange = lineRange(for: queryRange)
+        let resultText = substring(with: resultRange)
+
+        return (resultRange, resultText)
+    }
+
+    /// Returns the Range of the List Marker: This also includes NSTextAttachments
```

Comment:
What does "This also includes NSTextAttachments" mean?  Is it referring to the NSTextAttachment that is part of the .listMarkers array?

Reply:
Ahhhh forgot to patch that. Old code did not support indentation of Lists that had a TextAttachment (that is, only lists with a text based bullet were being indented).

In this PR that was changed, and we just support all the things. Patching up!